### PR TITLE
Use a single list of platforms to crossbuild

### LINKED ISSUE #3: Use a single list of platforms to crossbuild
then translate it to use the right `prom/golang-builder` image for each platforms
```
crossbuild:
    platforms:
        main:
            - linux/amd64
            - linux/386
            - darwin/amd64
            - darwin/386
            - windows/amd64
            - windows/386
        arm:
            - linux/arm
            - linux/arm64
        powerpc:
            - linux/ppc64
            - linux/ppc64le
```
to
```
crossbuild:
    platforms:
        - linux/amd64
        - linux/386
        - darwin/amd64
        - darwin/386
        - windows/amd64
        - windows/386
        - linux/arm
        - linux/arm64
        - linux/ppc64
        - linux/ppc64le
```

### PR DESCRIPTION
**Pull request by sdurrheimer**
Fixes #3

### CODE REVIEW

File: cmd/crossbuild.go [hunk 1]
```diff
@@ -104,19 +94,29 @@ func runCrossbuild() {
...
+		platforms = viper.GetStringSlice("crossbuild.platforms")
 
 		dockerMainBuilderImage    = fmt.Sprintf("%s:%s-main", dockerBuilderImageName, goVersion)
 		dockerARMBuilderImage     = fmt.Sprintf("%s:%s-arm", dockerBuilderImageName, goVersion)
 		dockerPowerPCBuilderImage = fmt.Sprintf("%s:%s-powerpc", dockerBuilderImageName, goVersion)
 		dockerMIPSBuilderImage    = fmt.Sprintf("%s:%s-mips", dockerBuilderImageName, goVersion)
 	)
 
+	for _, platform := range platforms {
+		for platformGroup, platformGroupToBuild := range map[*[]string]*[]string{&defaultMainPlatforms: &mainPlatforms, &defaultARMPlatforms: &ARMPlatforms, &defaultPowerPCPlatforms: &powerPCPlatforms, &defaultMIPSPlatforms: &MIPSPlatforms} {
```
Comment (grobie) [inline cmd/crossbuild.go]:
```go
for platformGroup, platformGroupToBuild := range map[*[]string]*[]string{
    &defaultMainPlatforms: &mainPlatforms,
    &defaultARMPlatforms: &ARMPlatforms,
    &defaultPowerPCPlatforms: &powerPCPlatforms,
    &defaultMIPSPlatforms: &MIPSPlatforms
} {
    // ...
}
```

Comment (grobie) [inline cmd/crossbuild.go]: I'd not pass string slice pointers, what is the reason for that? https://blog.golang.org/go-slices-usage-and-internals

File: cmd/crossbuild.go [hunk 2]
```diff
@@ -135,7 +135,7 @@ func runCrossbuild() {
 			dockerPowerPCBuilderImage, "-i", repoPath, "-p", q(powerPCPlatformsParam))
 	}
 
-	if MIPSPlatformsParam := strings.Join(MIPSPlatforms[:], " "); MIPSPlatformsParam != "" {
+	if MIPSPlatformsParam := strings.Join(MIPSPlatforms[:], " "); MIPSPlatformsParam != "" && version.Compare(goVersion, "1.6", ">=") {
```
Comment (grobie) [inline cmd/crossbuild.go]: So the minimum version required for mips is 1.6? Shouldn't that be communicated to the user when they try to build a project with go1.5 and specify mips?

Comment (sdurrheimer) [inline cmd/crossbuild.go]: We could, I also wanted to warn people when they specify a unknown/unhandled platform, but I didn't find a proper way to do it along side the platforms' matching.

**Comment by sdurrheimer** [comment]
I've updated the PR, getting rid of slice pointers + warns about unknown/unhandled platforms + warns about trying to build `MIPS` architectures with Go 1.5

File: cmd/crossbuild.go [hunk 3]
```diff
@@ -136,8 +148,12 @@ func runCrossbuild() {
 	}
 
 	if MIPSPlatformsParam := strings.Join(MIPSPlatforms[:], " "); MIPSPlatformsParam != "" {
```
Comment (grobie) [inline cmd/crossbuild.go]: Let's use lowercase names for local variables: `mipsPlatformsParam`.

**Comment by grobie** [comment]
:+1: Looks good so far. One more nitpick comment, but we can iterate on any further changes later. LGTM.

File: cmd/crossbuild.go (hunk 3 above)
Comment (sdurrheimer) [inline cmd/crossbuild.go]: I thought `MIPS` was a initialisms and should be in uppercase

Comment (grobie) [inline cmd/crossbuild.go]: if the first letter would be lowercase, it's fine to keep all lowercase. Just don't use `Mips`.

https://github.com/golang/go/wiki/CodeReviewComments#initialisms

Comment (sdurrheimer) [inline cmd/crossbuild.go]: `MIPS` is originally an acronym for `Microprocessor without Interlocked Pipeline Stages`, so I'm in favor to leave it in uppercase like ARM.

Comment (grobie) [inline cmd/crossbuild.go]: Uppercase variables are exported. Internal variables should not be exported, so they should not start with uppercase. The ARM variable should be adjusted as well `armPlatforms`.

Alternatively, change the variable name and start with something else in lowercase. For example `paramARMPlatform` if you prefer using all caps for the platform name.

Comment (sdurrheimer) [inline cmd/crossbuild.go]: Fixed

**Pull request merged**